### PR TITLE
PLATFORM-9311 Move AdEngine builds to Jenkinsfiles + Add Change Tracking

### PR DESCRIPTION
**Pull request by abador**

### CODE REVIEW

File: jenkins/functions.groovy [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      [
+          environment: environment,
+          issueKey   : issueKey,
+          success    : success
+      ]
+  )
+}
+
+def notifyChangesForBranch(boolean success, String branch, String buildUrl) {
+  def slackChannel = '#abador'
```
Comment (abador) [inline jenkins/functions.groovy:75]: If initial tests are fine we should change this to
```suggestion
  def slackChannel = '#adeng-release'
```

**Review by slayful — APPROVED** [review]

File: jenkins/functions.groovy [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  def message = success \
+      ? "New version of ad engine released <https://github.com/Wikia/ad-engine/tree/v$version|v$version>"
+      : ":x: <!here> Something went wrong during *ad-engine* release (<${buildUrl}|show log>)"
+  slackSend channel: slackChannel, message: message
+}
+
+def publishEngineFromBranch(String branch) {
+  sh("""
+      cd \$(pwd)/dist
+      S3_PATH="s3://fandom-ae-assets/platforms"
```
Comment (szymon-nawrocki) [inline jenkins/functions.groovy:93]:
```suggestion
      S3_PATH="s3://fandom-ae-assets/platforms/dev"
```

File: jenkins/Jenkinsfile_deploy.groovy [hunk 3]
```diff
@@ -0,0 +1,64 @@
+@Library('GlobalSharedLibrary') _
+
+def branch = params.branch
+def pullRequestNumber = params.pull_num
+def dryRun = params.dry_run
```
Comment (szymon-nawrocki) [inline jenkins/Jenkinsfile_deploy.groovy:5]: Comment for not forgetting to remove before merge :)

**Review by szymon-nawrocki — CHANGES_REQUESTED** [review]
Please change the s3 path for branch deployment

**Review by psmagiel — CHANGES_REQUESTED** [review]
Could we include running e2e tests after deploy to s3 dev succeeds, when the $commit is provided? https://github.com/Wikia/jenkins-jobs/blob/master/jobs/ads/ad-engine-platforms-deploy-branch.yaml

**Review by harnash — APPROVED** [review]

**Review by adpaste — APPROVED** [review]

**Review by adpaste — DISMISSED** [review]

File: jenkins/Jenkinsfile_deploy.groovy (hunk 3 above)
Comment (adpaste) [reply jenkins/Jenkinsfile_deploy.groovy]: Done.

**Review by psmagiel — APPROVED** [review]

File: jenkins/Jenkinsfile_deploy_branch.groovy [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    } else {
+        println "Can't mark release as unsuccessful"
+    }
+    throw e
+  } finally {
+    stage('Track change finish') {
+      def success = currentBuild.currentResult == 'SUCCESS'
+      sharedFunctions.markReleaseFinish(success, issueKey, environment)
+      sharedFunctions.notifyChangesForBranch(success, branch, buildUrl)
+
```
Comment (szymon-nawrocki) [inline jenkins/Jenkinsfile_deploy_branch.groovy:66]:
```suggestion
```

File: jenkins/Jenkinsfile_deploy_branch.groovy [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      stage("Track change start") {
+        issueKey = sharedFunctions.markReleaseStart(environment, version, branch, buildUrl)
+      }
+
+      stage('Deploy Version') {
+        sharedFunctions.publishEngineFromBranch(branch)
+      }
+
+      stage('Trigger tests') {
+        if (commit != "") {
```
[the remark this answers is not in the document]
Comment (szymon-nawrocki) [reply jenkins/Jenkinsfile_deploy_branch.groovy:44]: Okay, now I see that the commit is needed as a test parameter 🤔
Anyway it is not very intuitive that when I want to run the tests I need to provide a commit

[the remark this answers is not in the document]
Comment (abador) [reply jenkins/Jenkinsfile_deploy_branch.groovy:44]: It's the same in the original pipeline https://github.com/Wikia/jenkins-jobs/pull/1382/files#diff-495b402a5f8bece9acb97aa4f906a7acda37a739b6b6fbeb62a3dceb530141ff

**Review by szymon-nawrocki — APPROVED** [review]

**Pull request merged**